server/src/main/java/org/elasticsearch/common/CheckedTriConsumer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ *
+ */
+
+package org.elasticsearch.common;
+
+/**
+ * Represents an operation that accepts three arguments, returns no result, and allows throwing checked exceptions.
+ *
+ * @param <S> the type of the first argument
+ * @param <T> the type of the second argument
+ * @param <U> the type of the third argument
+ * @param <E> the type of anticipated exception
+ */
+@FunctionalInterface
+public interface CheckedTriConsumer<S, T, U, E extends Exception> {
```

Comment:
I don't know if i like this. IMO when we are at this number of parameters we could benefit from names and introduce a specific interface where it is needed (so in this case `SplitRequestExecutor` or something). It can still be a `@FunctionalInterface`.